### PR TITLE
[backend] add query to fetch documents

### PR DESCRIPTION
**Pull request by Christian-DONGMO**
### Proposed changes

*
*

### Related issues

*
*

### Checklist

- [x ] I consider the submitted work as finished
- [ ] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x ] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: openbas-model/src/main/java/io/openbas/database/raw/RawDocument.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package io.openbas.database.raw;
+
+import java.util.List;
+
+public interface RawDocument {
+
+    String getDocument_Id();
```
Comment (RomuDeuxfois) [inline openbas-model/src/main/java/io/openbas/database/raw/RawDocument.java:7]: Should be getDocument_id

File: openbas-model/src/main/java/io/openbas/database/repository/DocumentRepository.java [hunk 2]
```diff
@@ -30,4 +31,37 @@ public interface DocumentRepository extends CrudRepository<Document, String>, Jp
...
+            "array_remove(array_agg(sc.scenario_id), NULL) as document_scenarios " +
+            "from documents d left join exercises_documents exdoc on d.document_id = exdoc.document_id " +
+            "left join exercises ex on ex.exercise_id = exdoc.exercise_id " +
+            "left join scenarios_documents scdoc on d.document_id = scdoc.document_id " +
+            "left join scenarios sc on sc.scenario_id = scdoc.scenario_id " +
+            "left join documents_tags tagdoc on d.document_id = tagdoc.document_id " +
+            "left join tags tg on tg.tag_id = tagdoc.tag_id " +
+            "group by d.document_id " +
+            "order by document_id desc ", nativeQuery = true  )
+    List<RawDocument> findAllDocuments();
```
Comment (RomuDeuxfois) [inline openbas-model/src/main/java/io/openbas/database/repository/DocumentRepository.java:46]: Can you stick to the nomenclature?
rawAllDocuments()

and rawAllDocumentsByUser()

findAll is reserved for Document

**Review by RomuDeuxfois — CHANGES_REQUESTED** [review]

File: openbas-model/src/main/java/io/openbas/database/raw/RawDocument.java (hunk 1 above)
Comment (Christian-DONGMO) [reply openbas-model/src/main/java/io/openbas/database/raw/RawDocument.java:7]: It's done

File: openbas-model/src/main/java/io/openbas/database/repository/DocumentRepository.java (hunk 2 above)
Comment (Christian-DONGMO) [reply openbas-model/src/main/java/io/openbas/database/repository/DocumentRepository.java:46]: Roger that

**Review by RomuDeuxfois — APPROVED** [review]

**Pull request merged**